### ISSUE TITLE
htmlspecialChars() Warning

### ISSUE THREAD

**Issue by Shwetha9** [pattern-lab/patternlab-php#283]
Hi... not sure why i'm getting this error: Warning: htmlspecialchars() expects parameter 1 to be string... has anyone faced this before?

**Comment by dmolsen** [comment]
Do you have a line number for this error? That will help me track down what the issue might be. Thanks.

**Comment by Shwetha9** [comment]
The following are the line numbers: 41, 38, 20, 52. Thanks!

**Comment by dmolsen** [comment]
I was an idiot and didn't access for file name. Figured it would have been fairly straightforward since I only use `htmlspecialchars()` in a few places. Do you mind just copying and pasting the full error please? Thank you :)

**Comment by Shwetha9** [comment]
Sure! Full error: Warning: htmlspecialchars() expects parameter 1 to be string, array given in /Users/shwetha.anand/Desktop/patternlab-BAB/core/lib/Mustache/Engine.php(583) : eval()'d code on line 52

**Comment by Shwetha9** [comment]
One other issue- sorry :-D... Any reason why the AutoWatcherReload is not picking up changes to the CSS file? Only changes made to HTML files are getting picked up by the watcher... I've double checked the docs on your site and can ensure that none of the files have a '_' in front of it and are all stored inside the folder 'patterns'.

**Comment by Founts** [comment]
@Shwetha9 I came across this github issue when I had a similar error. For me, I resolved this by checking the data.json (and homepage.json) variable names - I was using "name" when it was already in use as a variable section in data.json. Hopefully this helps someone.

**Comment by Shwetha9** [comment]
@Founts Cheers for sharing that !

**Comment by Founts** [comment]
Ran into this again. The difficulty in troubleshooting this is you receive a message which says something like "Warning: htmlspecialchars() expects parameter 1 to be string, array given in /Users/jasonfounts/Documents/Projects/norcal-pl-v1/core/lib/Mustache/Engine.php(583) : eval()'d code on line 18" which points to a line number which isn't much help in the case I explained above. I'm not sure what line 18 is, but my problematic variable in data.json was line 38 and line 3 of my mustache file.

The real help in this message is "...expects parameter 1 to be string, array given..." since I used {{name}} in my mustache file which is an array in data.json. I shouldn't have used that data name. I hope this is helpful...
